src/wp-includes/post.php

```diff
@@ -1143,7 +1143,11 @@ function get_post_mime_type( $post = null ) {
  * @return string|false Post status on success, false on failure.
  */
 function get_post_status( $post = null ) {
-	$post = get_post( $post );
+	if ( is_object( $post ) && isset( $post->filter ) && 'sample' === $post->filter ) {
+		// Skip normalization
```

Comment:
@mcsf Is there a reason for the empty "if". What if we just invert the condition?

Reply:
👌 I don't mind either.

Reply:
Just a small suggestion (untested): If this code is staying (see above comments) perhaps can do `$post instanceof WP_Post` instead of `is_object( $post )` (seems a bit better?) and maybe add `$post->post_type === 'auto-draft'` to the conditional?

Reply:
I can only apply the first part of the suggestion. The post status is forced to "publish" for these calls.